src/uct/base/uct_iface.h

```diff
@@ -256,6 +256,21 @@ enum {
 UCS_CLASS_DECLARE(uct_iface_h, uct_iface_ops_t, uct_md_h);
 UCS_CLASS_DECLARE(uct_ep_t, uct_iface_h);
 
+/**
+ * 
+ */
+#define UCT_IFACE_FILL_INFO_STR_BUF(params, fmt, ...) do { \
```

Comment:
can it be a function instead of a macro?

Reply:
I originally made it a function but Leonid suggested otherwise (https://github.com/openucx/ucx/pull/9995#discussion_r1674333725). I am actually not sure what is preferable here and if the advantages of using a macro are worth it this time because it will not be called many times and it's a slow path so maybe we don't need to limit function calls that much

Reply:
the advantage of function is easier to debug/maintain, analyze backtrace, etc. since it is not fast path macro does not have an advantage IMO, function with `...` is simple enough, @gleon99 ?